core/relay/adaptor/text-embeddings-inference/adaptor.go

```diff
@@ -0,0 +1,175 @@
+package textembeddingsinference
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/labring/aiproxy/core/model"
+	"github.com/labring/aiproxy/core/relay/adaptor/openai"
+	"github.com/labring/aiproxy/core/relay/meta"
+	"github.com/labring/aiproxy/core/relay/mode"
+	relaymodel "github.com/labring/aiproxy/core/relay/model"
+	"github.com/labring/aiproxy/core/relay/utils"
+)
+
+// text-embeddings-inference adaptor supports rerank and embeddings models deployed by https://github.com/huggingface/text-embeddings-inference
+type Adaptor struct {
+	openai.Adaptor
+}
+
+// base url for text-embeddings-inference, fake
+const baseURL = "https://api.text-embeddings.net/v1"
+
+func (a *Adaptor) GetBaseURL() string {
+	return baseURL
+}
+
+func (a *Adaptor) GetModelList() []*model.ModelConfig {
+	return ModelList
+}
+
+func (a *Adaptor) GetRequestURL(meta *meta.Meta) (string, error) {
+	switch meta.Mode {
+	case mode.Rerank:
+		return a.GetBaseURL() + "/rerank", nil
+	case mode.Embeddings:
+		return a.GetBaseURL() + "/embeddings", nil
+	default:
+		return "", fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+}
+
+// text-embeddings-inference api see https://huggingface.github.io/text-embeddings-inference/#/Text%20Embeddings%20Inference/rerank
+
+func (a *Adaptor) SetupRequestHeader(meta *meta.Meta, _ *gin.Context, req *http.Request) error {
+	switch meta.Mode {
+	case mode.Rerank:
+		req.Header.Set("Content-Type", "application/json")
+	case mode.Embeddings:
+		req.Header.Set("Content-Type", "application/json")
+	default:
+		return fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+	return nil
+}
+
+func (a *Adaptor) ConvertRequest(meta *meta.Meta, req *http.Request) (string, http.Header, io.Reader, error) {
+	switch meta.Mode {
+	case mode.Rerank:
+		// Read the request body
+		body, err := io.ReadAll(req.Body)
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to read request body: %w", err)
+		}
+
+		// Parse the request body
+		var requestBody map[string]interface{}
+		if err := json.Unmarshal(body, &requestBody); err != nil {
+			return "", nil, nil, fmt.Errorf("failed to parse request body: %w", err)
+		}
+
+		// Create the rerank request, setting model is not supported
+		rerankRequest := map[string]interface{}{
+			// "model": meta.OriginModel,
+			"query": requestBody["query"],
+			"texts": requestBody["documents"],
+		}
```

Comment:
Use the sonic ast library to convert only the documents field to the texts field, while retaining other parameters